src/types/entityTypes.ts

```diff
@@ -20,3 +20,82 @@ export interface EntityContext<T> extends InvocationContext {
      */
     df: DurableEntityContext<T>;
 }
+
+/**
+ * Provides functionality for application code implementing an entity operation.
+ * @param TState the JSON-serializable type of the Entity's state
+ */
+export interface DurableEntityContext<TState> {
```

Comment:
I noticed this used to be an `export type` but now is `export interface`. Can you please clarify the difference?
Also, why does this export an `interface` where the orchestrator-context exports a `class`?

Reply:
For this use-case, there are no relevant differences. I'm not sure what the subtle differences between `type` and `interface` in TypeScript (at least `>2.1`) are, but for defining object/class structure, they're effectively the same. However, it feels to me at least that using `interface` for this use-case is more idiomatic, when defining the structure of an object. I think the same also applies to the earlier `DurableClient` and `DurableOrchestrationContext`, and if it was up to me, I'd rather change those to interfaces too. One argument is that you, as the user, _shouldn't_ be calling the constructor of those types. However, changing classes to interfaces _would_ be a breaking change, which is why I haven't done that here, but changing a `type` to an `interface` isn't.

Reply:
I agree with David's sentiment here and would prefer if we avoid changes like this during two scenarios:
1. When dealing with our public contract. If the risk and the value are equal (both low in this case), I err on the side of caution
2. During PRs that are mostly just moving code. These PRs are difficult to review, so even minor changes should be avoided

That being said, I don't have a specific example of when switching a type to an interface would cause a breaking change so I won't block on this PR. Please consider this feedback in future PRs, though

Reply:
Sounds good, I appreciate the feedback, thank you!